### PR TITLE
KYLO-2423 Invalidate InitializeFeed cache when feed deleted

### PR DESCRIPTION
**Pull request by BoyingLu**
The 'mvn test' has been passed and the patch has been verified in my test environment

### CODE REVIEW

File: integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java [hunk 1]
```diff
@@ -140,10 +155,13 @@
 
     private Map<String, AtomicInteger> retryCounts = Collections.synchronizedMap(new HashMap<>());
 
+    private CleanupEventService cleanupEventService;
+
     @OnScheduled
     public void scheduled(ProcessContext context) {
```
Comment (KyloTech) [inline integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java]: MAJOR Add the "@Override" annotation above this method signature [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1161)

File: integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java [hunk 2]
```diff
@@ -127,6 +135,13 @@
...
         .build();
 
+    public static final PropertyDescriptor CLEANUP_SERVICE = new PropertyDescriptor.Builder()
+        .name("Feed Cleanup Event Service")
+        .description("Service that manages the cleanup of feeds.")
+        .identifiesControllerService(CleanupEventService.class)
+        .required(true)
+        .build();
+
     Relationship REL_INITIALIZE = new Relationship.Builder()
```
Comment (KyloTech) [inline integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java]: MINOR Rename this field "REL_INITIALIZE" to match the regular expression '^[a-z][a-zA-Z0-9]*$'. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS00116)

File: integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java [hunk 3]
```diff
@@ -57,13 +62,16 @@
...
+import static com.thinkbiganalytics.nifi.v2.common.CommonProperties.FEED_NAME;
+
 /**
  */
 @EventDriven
 @InputRequirement(InputRequirement.Requirement.INPUT_ALLOWED)
 @Tags({"feed", "initialize", "initialization", "thinkbig"})
 @CapabilityDescription("Controls setup of a feed by routing to an initialization or re-initialization flow.")
-public class InitializeFeed extends FeedProcessor {
+public class InitializeFeed extends FeedProcessor implements CleanupListener {
```
Comment (KyloTech) [inline integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java]: MAJOR This class has 7 parents which is greater than 5 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
MINOR Override the "equals" method in this class. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2160)

**Comment by KyloTech** [comment]
SonarQube analysis reported 31 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 17 major
* ![MINOR][MINOR] 11 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [InitializeFeed.java#L80](https://github.com/Teradata/kylo/blob/fc3eaa484dee346f1a855f187c54130c395c7a47/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L80): Define a constant instead of duplicating this literal "RETRY" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [FeedProcessor.java#L12](https://github.com/Teradata/kylo/blob/fc3eaa484dee346f1a855f187c54130c395c7a47/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/common/FeedProcessor.java#L12): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [FeedProcessor.java#L55](https://github.com/Teradata/kylo/blob/fc3eaa484dee346f1a855f187c54130c395c7a47/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/common/FeedProcessor.java#L55): This class has 6 parents which is greater than 5 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L12](https://github.com/Teradata/kylo/blob/fc3eaa484dee346f1a855f187c54130c395c7a47/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L12): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L273](https://github.com/Teradata/kylo/blob/fc3eaa484dee346f1a855f187c54130c395c7a47/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L273): Remove these unused method parameters. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L279](https://github.com/Teradata/kylo/blob/fc3eaa484dee346f1a855f187c54130c395c7a47/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L279): Call "Optional#isPresent()" before accessing the value. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L309](https://github.com/Teradata/kylo/blob/fc3eaa484dee346f1a855f187c54130c395c7a47/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L309): Remove these unused method parameters. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L315](https://github.com/Teradata/kylo/blob/fc3eaa484dee346f1a855f187c54130c395c7a47/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L315): Call "Optional#isPresent()" before accessing the value. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L332](https://github.com/Teradata/kylo/blob/fc3eaa484dee346f1a855f187c54130c395c7a47/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L332): Use "Boolean.toString" instead. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2131)
1. ![MAJOR][MAJOR] [MetadataRecorder.java#L9](https://github.com/Teradata/kylo/blob/fc3eaa484dee346f1a855f187c54130c395c7a47/integrations/nifi/nifi-nar-bundles/nifi-standard-services/nifi-core-service-api/src/main/java/com/thinkbiganalytics/nifi/core/api/metadata/MetadataRecorder.java#L9): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by KyloTech** [comment]
SonarQube analysis reported 31 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 17 major
* ![MINOR][MINOR] 11 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [InitializeFeed.java#L80](https://github.com/Teradata/kylo/blob/99993ba05b4f44b8527c63f793f1d38d454d2f7b/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L80): Define a constant instead of duplicating this literal "RETRY" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [FeedProcessor.java#L12](https://github.com/Teradata/kylo/blob/99993ba05b4f44b8527c63f793f1d38d454d2f7b/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/common/FeedProcessor.java#L12): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [FeedProcessor.java#L55](https://github.com/Teradata/kylo/blob/99993ba05b4f44b8527c63f793f1d38d454d2f7b/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/common/FeedProcessor.java#L55): This class has 6 parents which is greater than 5 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L12](https://github.com/Teradata/kylo/blob/99993ba05b4f44b8527c63f793f1d38d454d2f7b/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L12): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L273](https://github.com/Teradata/kylo/blob/99993ba05b4f44b8527c63f793f1d38d454d2f7b/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L273): Remove these unused method parameters. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L279](https://github.com/Teradata/kylo/blob/99993ba05b4f44b8527c63f793f1d38d454d2f7b/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L279): Call "Optional#isPresent()" before accessing the value. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L309](https://github.com/Teradata/kylo/blob/99993ba05b4f44b8527c63f793f1d38d454d2f7b/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L309): Remove these unused method parameters. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L315](https://github.com/Teradata/kylo/blob/99993ba05b4f44b8527c63f793f1d38d454d2f7b/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L315): Call "Optional#isPresent()" before accessing the value. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L332](https://github.com/Teradata/kylo/blob/99993ba05b4f44b8527c63f793f1d38d454d2f7b/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L332): Use "Boolean.toString" instead. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2131)
1. ![MAJOR][MAJOR] [MetadataRecorder.java#L9](https://github.com/Teradata/kylo/blob/99993ba05b4f44b8527c63f793f1d38d454d2f7b/integrations/nifi/nifi-nar-bundles/nifi-standard-services/nifi-core-service-api/src/main/java/com/thinkbiganalytics/nifi/core/api/metadata/MetadataRecorder.java#L9): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by KyloTech** [comment]
SonarQube analysis reported 31 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 17 major
* ![MINOR][MINOR] 11 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [InitializeFeed.java#L80](https://github.com/Teradata/kylo/blob/612e1eca599a10e0106108a353d8d65f2116e3aa/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L80): Define a constant instead of duplicating this literal "RETRY" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [FeedProcessor.java#L12](https://github.com/Teradata/kylo/blob/612e1eca599a10e0106108a353d8d65f2116e3aa/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/common/FeedProcessor.java#L12): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [FeedProcessor.java#L55](https://github.com/Teradata/kylo/blob/612e1eca599a10e0106108a353d8d65f2116e3aa/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/common/FeedProcessor.java#L55): This class has 6 parents which is greater than 5 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L12](https://github.com/Teradata/kylo/blob/612e1eca599a10e0106108a353d8d65f2116e3aa/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L12): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L273](https://github.com/Teradata/kylo/blob/612e1eca599a10e0106108a353d8d65f2116e3aa/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L273): Remove these unused method parameters. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L279](https://github.com/Teradata/kylo/blob/612e1eca599a10e0106108a353d8d65f2116e3aa/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L279): Call "Optional#isPresent()" before accessing the value. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L309](https://github.com/Teradata/kylo/blob/612e1eca599a10e0106108a353d8d65f2116e3aa/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L309): Remove these unused method parameters. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L315](https://github.com/Teradata/kylo/blob/612e1eca599a10e0106108a353d8d65f2116e3aa/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L315): Call "Optional#isPresent()" before accessing the value. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L332](https://github.com/Teradata/kylo/blob/612e1eca599a10e0106108a353d8d65f2116e3aa/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L332): Use "Boolean.toString" instead. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2131)
1. ![MAJOR][MAJOR] [MetadataRecorder.java#L9](https://github.com/Teradata/kylo/blob/612e1eca599a10e0106108a353d8d65f2116e3aa/integrations/nifi/nifi-nar-bundles/nifi-standard-services/nifi-core-service-api/src/main/java/com/thinkbiganalytics/nifi/core/api/metadata/MetadataRecorder.java#L9): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by KyloTech** [comment]
SonarQube analysis reported 31 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 17 major
* ![MINOR][MINOR] 11 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [InitializeFeed.java#L80](https://github.com/Teradata/kylo/blob/4cd156eb2f4ca66b2978416e299020d7aa871dbd/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L80): Define a constant instead of duplicating this literal "RETRY" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [FeedProcessor.java#L12](https://github.com/Teradata/kylo/blob/4cd156eb2f4ca66b2978416e299020d7aa871dbd/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/common/FeedProcessor.java#L12): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [FeedProcessor.java#L55](https://github.com/Teradata/kylo/blob/4cd156eb2f4ca66b2978416e299020d7aa871dbd/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/common/FeedProcessor.java#L55): This class has 6 parents which is greater than 5 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L12](https://github.com/Teradata/kylo/blob/4cd156eb2f4ca66b2978416e299020d7aa871dbd/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L12): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L273](https://github.com/Teradata/kylo/blob/4cd156eb2f4ca66b2978416e299020d7aa871dbd/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L273): Remove these unused method parameters. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L279](https://github.com/Teradata/kylo/blob/4cd156eb2f4ca66b2978416e299020d7aa871dbd/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L279): Call "Optional#isPresent()" before accessing the value. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L309](https://github.com/Teradata/kylo/blob/4cd156eb2f4ca66b2978416e299020d7aa871dbd/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L309): Remove these unused method parameters. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L315](https://github.com/Teradata/kylo/blob/4cd156eb2f4ca66b2978416e299020d7aa871dbd/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L315): Call "Optional#isPresent()" before accessing the value. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L332](https://github.com/Teradata/kylo/blob/4cd156eb2f4ca66b2978416e299020d7aa871dbd/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L332): Use "Boolean.toString" instead. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2131)
1. ![MAJOR][MAJOR] [MetadataRecorder.java#L9](https://github.com/Teradata/kylo/blob/4cd156eb2f4ca66b2978416e299020d7aa871dbd/integrations/nifi/nifi-nar-bundles/nifi-standard-services/nifi-core-service-api/src/main/java/com/thinkbiganalytics/nifi/core/api/metadata/MetadataRecorder.java#L9): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by KyloTech** [comment]
SonarQube analysis reported 31 issues
* ![CRITICAL][CRITICAL] 1 critical
* ![MAJOR][MAJOR] 17 major
* ![MINOR][MINOR] 11 minor
* ![INFO][INFO] 2 info

Watch the comments in this conversation to review them.

#### Top 10 extra issues

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![CRITICAL][CRITICAL] [InitializeFeed.java#L80](https://github.com/Teradata/kylo/blob/9bdb04408f58ed639b8001baf874f4e36b76cc3d/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L80): Define a constant instead of duplicating this literal "RETRY" 3 times. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1192)
1. ![MAJOR][MAJOR] [FeedProcessor.java#L12](https://github.com/Teradata/kylo/blob/9bdb04408f58ed639b8001baf874f4e36b76cc3d/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/common/FeedProcessor.java#L12): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [FeedProcessor.java#L55](https://github.com/Teradata/kylo/blob/9bdb04408f58ed639b8001baf874f4e36b76cc3d/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/common/FeedProcessor.java#L55): This class has 6 parents which is greater than 5 authorized. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AMaximumInheritanceDepth)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L12](https://github.com/Teradata/kylo/blob/9bdb04408f58ed639b8001baf874f4e36b76cc3d/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L12): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L273](https://github.com/Teradata/kylo/blob/9bdb04408f58ed639b8001baf874f4e36b76cc3d/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L273): Remove these unused method parameters. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L279](https://github.com/Teradata/kylo/blob/9bdb04408f58ed639b8001baf874f4e36b76cc3d/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L279): Call "Optional#isPresent()" before accessing the value. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L309](https://github.com/Teradata/kylo/blob/9bdb04408f58ed639b8001baf874f4e36b76cc3d/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L309): Remove these unused method parameters. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1172)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L315](https://github.com/Teradata/kylo/blob/9bdb04408f58ed639b8001baf874f4e36b76cc3d/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L315): Call "Optional#isPresent()" before accessing the value. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS3655)
1. ![MAJOR][MAJOR] [InitializeFeed.java#L332](https://github.com/Teradata/kylo/blob/9bdb04408f58ed639b8001baf874f4e36b76cc3d/integrations/nifi/nifi-nar-bundles/nifi-core-bundle/nifi-core-processors/src/main/java/com/thinkbiganalytics/nifi/v2/init/InitializeFeed.java#L332): Use "Boolean.toString" instead. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3AS2131)
1. ![MAJOR][MAJOR] [MetadataRecorder.java#L9](https://github.com/Teradata/kylo/blob/9bdb04408f58ed639b8001baf874f4e36b76cc3d/integrations/nifi/nifi-nar-bundles/nifi-standard-services/nifi-core-service-api/src/main/java/com/thinkbiganalytics/nifi/core/api/metadata/MetadataRecorder.java#L9): This block of commented-out lines of code should be removed. [rule](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)

[CRITICAL]: https://sonarsource.github.io/sonar-github/severity-critical.png 'Severity: CRITICAL'
[INFO]: https://sonarsource.github.io/sonar-github/severity-info.png 'Severity: INFO'
[MAJOR]: https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR'
[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

**Comment by felten** [comment]
Thank you for this pull request but it is unusable because it is based off your "master" branch rather than a topic branch, and has rebase conflicts.  It will likely always have conflicts because it is based off of master.  

It currently has 36 commits in it with only a few related to fixing KYLO-2423.  When you are making changes for a specific issue you should create a topic branch containing only those commits relevant to that issue and create the pull request from that topic branch.  Please see: https://help.github.com/articles/creating-a-pull-request

I have been able to reproduce the issue and see where the problem is.  Your solution is a good one but I am considering simply moving the feed ID caching into the metadata service rather than keeping it at the FeedProcessor level.  That way the caching is done at one place and only the metadata service need react to feed deletions.  If any problems come up with that approach we can fall back to your solution.  Feel free to recreate a new pull request if you like with only your KYLO-2423 changes in a topic branch if we need to use your solution.  I will be closing this pull request.

**Pull request closed without merging**